tests/test_operation.py

```diff
@@ -1475,3 +1500,19 @@ def test_composed(self):
         assert not both(self.cnot)
         assert not both(self.rot)
         assert not both(self.exp)
+
+
+class TestStaticProperties:
+    @pytest.mark.parametrize(
+        "op, num_params",
+        [
+            (qml.Identity, 0),
+            (qml.PauliZ, 0),
+            (qml.RY, 1),
+            (qml.TwoModeSqueezing, 2),
+        ],
+    )
+    def test_num_params_can_be_static(self, op, num_params):
+        """Test for soee of Operations which can declare num_params as a
+        class property that they actually do so"""
+        assert op.num_params == num_params
```

Comment:
I personally would delete this test. If you really want to know that these gates declare the num_params correctly, it should be tested in the gate unit tests!